### PR TITLE
Add sliding window high level impl convenience

### PR DESCRIPTION
**Pull request by Purg**
Adding another higher level implementation "convenience" for the sliding-window/difference-occlusion algorithms. This follows the same encapsulation method as the other RISE convenience implementation.

### CODE REVIEW

File: docs/release_notes/pending_release.rst [hunk 1]
```diff
@@ -51,10 +51,14 @@ Implementations
...
   de-biased form of RISE with an optional input parameter.
 
-* Add ``RISEStack`` implementation of ``GenerateImageClassifierBlackboxSaliency``
+* Add ``RISEStack`` implementation of the ``GenerateImageClassifierBlackboxSaliency``
   interface as a simple way to invoke the combination of RISE component
   algorithms.
 
+* Add ``SlidingWindowStack`` implementation of the ``GenerateImageClassifierBlackboxSaliency``
+  interface as a simple way to invoke the combination of the Sliding Window
+  perturbation method and the occlusion-difference based scoring method.
```
Comment (brianhhu) [inline docs/release_notes/pending_release.rst:60]: "occlusion-difference based scoring method" -> "occlusion-based scoring method" seems slightly less confusing.

File: xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from xaitk_saliency.impls.gen_image_classifier_blackbox_sal.occlusion_based import PerturbationOcclusion
+from xaitk_saliency.impls.perturb_image.sliding_window import SlidingWindow
+from xaitk_saliency.interfaces.gen_image_classifier_blackbox_sal import GenerateImageClassifierBlackboxSaliency
+
+
+class SlidingWindowStack (GenerateImageClassifierBlackboxSaliency):
+    """
+    Encapsulation of the perturbation-occlusion method using specifically
+    sliding windows and the occlusion-scoring method.
+    """
```
Comment (brianhhu) [inline xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py:16]: In our corresponding RISEStack method, we duplicated the docstring with information on the input parameters. However, I see that's this is missing here (e.g. info on window_size, stride, threads).

File: xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            threads=threads,
+        )
+
+    def _generate(self, ref_image: np.ndarray, blackbox: ClassifyImage) -> np.ndarray:
+        return self._po.generate(ref_image, blackbox)
+
+    @classmethod
+    def get_default_config(cls) -> Dict[str, Any]:
+        # Minor override following SlidingWindow impl because of list/tuple
+        # equality check differences.
```
Comment (brianhhu) [inline xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py:39]: Can you unpack this a bit more? Under what scenarios is this `@classmethod`  pattern required?

File: xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        cfg = super().get_default_config()
+        cfg.update(SlidingWindow.get_default_config())
+        return cfg
+
+    def get_config(self) -> Dict[str, Any]:
+        c = self._po.perturber.get_config()
+        c['threads'] = self._po.threads
```
Comment (brianhhu) [inline xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py:46]: Why do we have to explicitly set `threads` in the config here, but not in the corresponding RISEStack method?

**Review by brianhhu — APPROVED** [review]
LGTM, added some relatively minor review comments, mostly for clarification.

File: docs/release_notes/pending_release.rst (hunk 1 above)
Comment (Purg) [reply docs/release_notes/pending_release.rst:60]: Done.

File: xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py (hunk 2 above)
Comment (Purg) [reply xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py:16]: I indeed missed getting the parameter docs here. I will add them.

File: xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py (hunk 3 above)
Comment (Purg) [reply xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py:39]: I thought I would be "smart" and make user of the currying that the `SlidingWindow` implementation does, but I think its both less clear as well as "wrong" (uses defaults of `SlidingWindow`s constructor, which may be different than here).

File: xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py (hunk 4 above)
Comment (Purg) [reply xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py:46]: This has to do with what parameters are defined in the `perturber` instance that are also "shared" with this implementation. In the RISE case, the RISE perturbation implementation had a threading parameter that is set in the `RISEStack` constructor to both that impl *and* the `PerturbationOcclusion` constructor. In this case, with `SlidingWindowStack`, the perturber being drawn from does *not* have a threading parameter, which is only encoded in the `PerturbationOcclusion` instance.

**Comment by Purg** [comment]
@brianhhu PTAL at the latest updates here.

**Comment by brianhhu** [comment]
LGTM- thanks!

**Pull request merged**